examples/plot_nonstationary_error.py

```diff
@@ -0,0 +1,205 @@
+"""
+Non-stationarity of elevation measurement errors
+================================================
+
+Digital elevation models have a precision that can vary with terrain and instrument-related variables. However, quantifying
+this precision is complex and non-stationarities, i.e. variability of the measurement error, has rarely been
+accounted for, with only some studies that used arbitrary filtering thresholds on the slope or other variables (see :ref:`intro`).
+
+Quantifying the non-stationarities in elevation measurement errors is essential to use stable terrain as a proxy for
+assessing the precision on other types of terrain (Hugonnet et al., in prep) and allows to standardize the measurement
+errors to reach a stationary variance, an assumption necessary for spatial statistics (see :ref`spatialstats`).
+
+Here, we show an example in which we identify terrain-related non-stationarities for a DEM difference of Longyearbyen glacier.
+We quantify those non-stationarities by `binning <https://en.wikipedia.org/wiki/Data_binning>`_ robustly
+in N-dimension using :func:`xdem.spstats.nd_binning` and applying a N-dimensional interpolation
+:func:`xdem.spstats.interp_nd_binning` to estimate a numerical function of the measurement error and derive the spatial
+distribution of elevation measurement errors of the difference of DEMs.
+
+**Reference**: `Hugonnet et al. (2021) <https://doi.org/10.1038/s41586-021-03436-z>`_, applied to the terrain slope
+and quality of stereo-correlation (Equation 1, Extended Data Fig. 3a).
+"""
+# sphinx_gallery_thumbnail_number = 8
+import matplotlib.pyplot as plt
+import numpy as np
+import xdem
+import geoutils as gu
+
+# %%
+# We start by loading example files including a difference of DEMs at Longyearbyen glacier, the reference DEM later used to derive
+# several terrain attributes, and the outlines to rasterize a glacier mask.
+# Prior to differencing, the DEMs were aligned using :class:`xdem.coreg.NuthKaab` as shown in
+# the :ref:`sphx_glr_auto_examples_plot_nuth_kaab.py` example. We later refer to those elevation differences as *dh*.
+
+ref_dem = xdem.DEM(xdem.examples.get_path("longyearbyen_ref_dem"))
+ddem = xdem.DEM(xdem.examples.get_path("longyearbyen_ddem"))
+glacier_outlines = gu.Vector(xdem.examples.get_path("longyearbyen_glacier_outlines"))
+mask_glacier = glacier_outlines.create_mask(ddem)
+
+# %%
+# We remove values on unstable terrain
+ddem.data[mask_glacier] = np.nan
```

Comment:
This is not the way we should do it, as this will conflict with the actual nodata value used. But currently, masking pixels require changing both self.data.data and self.data.mask... I need to find a solution for this.

Reply:
Yes, a function very much like `get_array_and_mask` that works for all types.
Maybe `update_mask`?